### DISCUSSION TITLE
Why can not access 'type' in EdgeProps  when i create custom edge component using '@xyflow/react'?

### DISCUSSION THREAD

**Discussion by peanudge** [xyflow/xyflow#4595] [Q&A] [answered]
Hi, xyflow team :) First, thank you for making amazing graph draw library. it is very useful to me.

I could not access 'type' of edge during creating custom edge component like following.

```jsx
const CustomEdgeComponent = (
    props: EdgeProps<CustomEdgeType>
) => {
     const type = props.type; // Problem: EdgeProps do not allow to access 'type' data.
     ...
}
```

Question 1) Is there alternative way to get edge's type?

if no special reason about hiding `type` of `EdgeType`, 

I hope to add `Pick<EdgeType, ... | 'type'>` in following type definition for more customizable edge component

https://github.com/xyflow/xyflow/blob/9b768b2dae9f046c4c10772b267ad3116c3333a5/packages/react/src/types/edges.ts#L109

# Incompatible type mismatch

I'd like to share additional strange point.

`EdgeTypes` required `{type: any}` as props of  `<ReactFlow/>` component. this problem make hard to write strict typescript for me. 

https://github.com/xyflow/xyflow/blob/e55b2ad8d0d011ecf1bbc443feba2a55ba17700b/packages/react/src/types/general.ts#L42-L52

FYI) I tried to apply HOC pattern to CustomEdgeComponent for adding,removing edit ability in case by case

```tsx
import { EdgeTypes } from "'@xyflow/react"
const edgeTypes: EdgeTypes = {
    customEdge: withEditable(CustomEdge), // Typescript error b/c EdgeTypes required {data: any, type: any}
}

const CustomEdge = (props: EdgeProps) => {...}

type CustomEdgeComponentProps =   ComponentProps<typeof GraphEdgeComponent>
const withEditable = (EdgeComponent: ComponentType<CustomEdgeComponentProps>) => { ... }
```

Above code is summary in my situation. 
Also, I can resolve through hack like `type CustomEdgeComponentProps = ComponentProps<typeof GraphEdgeComponent> & {data:any, type:any}`.

But, i hoped graceful solution

**Comment by bcakmakoglu** [comment, marked as answer]
An alternative way to get the edge type would be to use [`getEdge`](https://reactflow.dev/api-reference/types/react-flow-instance#get-edge)

```ts
const edge = getEdge(edgeId)

const type = edge.type
